developerportal/apps/taskqueue/tasks.py

```diff
@@ -61,3 +71,48 @@ def selectively_invalidate_cdn():
         f"selected CDN keys: {selected_targets}"
     )
     invalidate_cdn(invalidation_targets=selected_targets)
+
+    # Warming the top-level pages will help, even if we don't warm them all
+    site = Site.objects.filter(is_default_site=True).get()  # Should fail loudly
+
+    proto = "https://" if site.port == 443 else "http://"
+    host = f"{proto}{site.hostname}"
+
+    urls_to_warm = [
+        f"{host}{x}" for x in [events_root_path, people_root_path, topics_root_path]
+    ]
+    selectively_warm_cdn.apply_async(
+        args=[urls_to_warm], countdown=DELAY_BEFORE_WARMING_CDN
+    )
+
+
+@app.task
+def warm_entire_cdn():
+    """For use after blanket invalidation, this function iterates through all
+    URLs listed in the sitemap.
+
+    Note that this can be several hundreds of pages, hence it needs to happen
+    in a background task.
+    """
+    urls = get_all_urls_from_sitemap()
+
+    return _warm_cdn(urls)
+
+
+@app.task
+def selectively_warm_cdn(urls):
+    return _warm_cdn(urls)
+
+
+def _warm_cdn(urls):
+
+    total_urls = len(urls)
+    logger.info(f"Warming CDN. {total_urls} URLs obtained from sitemap.")
+
+    for i, url in enumerate(urls):
+        logger.info(f"Requesting {i+1}/{total_urls}: {url}")
+        response = requests.get(url)
```

Comment:
If the number of pages gets too large, it might be worth chunking them down into batches and handling each batch as a separate task, with a [Celery chord](https://docs.celeryproject.org/en/latest/reference/celery.html#celery.chord) orchestrating them